src/Algebra/Graph/Relation/Symmetric.hs

```diff
@@ -11,27 +11,468 @@
+-- vertexList . 'vertices' == 'Data.List.nub' . 'Data.List.sort'
+-- @
+vertexList :: SymmetricRelation a -> [a]
+vertexList = R.vertexList . fromSymmetric
+
+-- | The sorted list of edges of a graph.
+-- Complexity: /O(n + m)/ time and /O(m)/ memory.
+--
+-- @
+-- edgeList 'empty'          == []
+-- edgeList ('vertex' x)     == []
+-- edgeList ('edge' x y)     == [(x,y)]
+-- edgeList ('star' 2 [3,1]) == [(2,1), (2,3)]
+-- @
+edgeList :: Ord a => SymmetricRelation a -> [(a, a)]
+edgeList = deduplicate . R.edgeList . fromSymmetric
+
+-- | The set of vertices of a given graph.
+-- Complexity: /O(1)/ time.
+--
+-- @
+-- vertexSet 'empty'      == Set.'Set.empty'
+-- vertexSet . 'vertex'   == Set.'Set.singleton'
+-- vertexSet . 'vertices' == Set.'Set.fromList'
+-- @
+vertexSet :: SymmetricRelation a -> Set.Set a
+vertexSet = R.vertexSet . fromSymmetric
+
+-- | The set of edges of a given graph.
+-- Complexity: /O(m)/ time.
+--
+-- @
+-- edgeSet 'empty'      == Set.'Set.empty'
+-- edgeSet ('vertex' x) == Set.'Set.empty'
+-- (edgeSet ('edge' x y) == Set.'Set.singleton' (x,y)) || (edgeSet ('edge' x y) == Set.'Set.singleton' (y,x))
+-- @
+edgeSet :: Ord a => SymmetricRelation a -> Set.Set (a, a)
+edgeSet = deduplicateSet . R.edgeSet . fromSymmetric
+
+-- | The sorted /adjacency list/ of a graph.
+-- Complexity: /O(n + m)/ time and /O(m)/ memory.
+--
+-- @
+-- adjacencyList 'empty'          == []
+-- adjacencyList ('vertex' x)     == [(x, [])]
+-- adjacencyList ('edge' 1 2)     == [(1, [2]), (2, [1])]
+-- adjacencyList ('star' 2 [3,1]) == [(1, [2]), (2, [1,3]), (3, [2])]
+-- 'stars' . adjacencyList        == id
+-- @
+adjacencyList :: Eq a => SymmetricRelation a -> [(a, [a])]
+adjacencyList = R.adjacencyList . fromSymmetric
+
+-- | The /path/ on a list of vertices.
+-- Complexity: /O((n + m) * log(n))/ time and /O(n + m)/ memory.
+--
+-- @
+-- path []        == 'empty'
+-- path [x]       == 'vertex' x
+-- path [x,y]     == 'edges' [(x,y)]
+-- path           == path . 'reverse'
+-- @
+path :: Ord a => [a] -> SymmetricRelation a
+path xs = case xs of []     -> empty
+                     [x]    -> vertex x
+                     (_:ys) -> edges (zip xs ys)
+
+-- | The /circuit/ on a list of vertices.
+-- Complexity: /O((n + m) * log(n))/ time and /O(n + m)/ memory.
+--
+-- @
+-- circuit []        == 'empty'
+-- circuit [x]       == 'edge' x x
+-- circuit [x,y]     == 'edges' [(x,y)]
+-- circuit [x,y,z]   == 'edges' [(x,y),(x,z),(y,z)]
+-- circuit           == circuit . 'reverse'
+-- @
+circuit :: Ord a => [a] -> SymmetricRelation a
+circuit []     = empty
+circuit (x:xs) = path $ [x] ++ xs ++ [x]
+
+-- | The /clique/ on a list of vertices.
+-- Complexity: /O((n + m) * log(n))/ time + /O(m*log(m)) time from computing the symmetricClosure and /O(n + m)/ memory.
+--
+-- @
+-- clique []         == 'empty'
+-- clique [x]        == 'vertex' x
+-- clique [x,y]      == 'edges' [(x,y)]
+-- clique [x,y,z]    == 'edges' [(x,y), (x,z), (y,z)]
```

Comment:
It's worth adding something like `clique . reverse == clique` here to emphasise that order does not matter.